src/components/pinnable.js

```diff
@@ -74,5 +74,12 @@ AFRAME.registerComponent("pinnable", {
     }
 
     this.wasHeld = held;
+
+    this.transformObjectSystem = this.transformObjectSystem || AFRAME.scenes[0].systems["transform-selected-object"];
+    const transforming = this.transformObjectSystem.transforming && this.transformObjectSystem.target.el === this.el;
+    if (!transforming && this.wasTransforming) {
+      this._fireEventsAndAnimate(this.data, true);
```

Comment:
are there cases where you were transforming the object and holding it,  and stop doing so at the same time? If so we would end up firing the events twice.. not sure if that's an issue.